### PR TITLE
CLangSMG class port

### PR DESCRIPTION
**Pull request by viktormalik**

### CODE REVIEW

**Comment by michkot** [comment]
TODO: change all custom exceptions to derive from runtime_error or logic_error instead of basic exception - these already contains needed message definition.

File: src/graphs/CLangStackFrame.cc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    to_return.insert(var.second);
+  }
+  return to_return;
+//  if (returnValueObject != null) {
+//    retset.add(returnValueObject);
+//  }
+}
+
+const std::string CLangStackFrame::GetFunctionDeclaration() const { return stack_function_; }
+
```
Comment (michkot) [inline src/graphs/CLangStackFrame.cc:63]: Are we gonna implement with custom type for Function Declaration like in ? https://github.com/petr-muller/smg/blob/master/src/main/java/cz/afri/smg/types/CFunctionDeclaration.java . I need to know for the SMGPlotter part.

Comment (viktormalik) [inline src/graphs/CLangStackFrame.cc:63]: I believe we should, I only used string as temporary replacement for other methods. But this is more question for PP, who is implementing types.

Comment (michkot) [inline src/graphs/CLangStackFrame.cc:63]: OK + Sure, will discuss that with him tomorrow.

**Pull request merged**